### PR TITLE
Update Queue API, documentation

### PR DESCRIPTION
**Pull request by aliabid94**
Made Queue API a little clearer, added docstrings and documentation to site. Remove print statements. Improved estimation logic (no need to store duration history as a list)

### CODE REVIEW

File: gradio/blocks.py [hunk 1]
```diff
@@ -768,28 +769,32 @@ def clear(self):
         self.children = []
         return self
 
-    def configure_queue(
+    @document()
+    def queue(
```
Comment (abidlabs) [inline gradio/blocks.py:773]: Very clean, love it!

File: gradio/blocks.py [hunk 2]
```diff
@@ -768,28 +769,32 @@ def clear(self):
...
-            update_intervals: Queue will send estimations to the clients in intervals=update_intervals when live_queue_updates==false
-            duration_history_size: Queue duration estimation calculation window size.
+            concurrency_count: Number of worker threads that will be processing requests concurrently.
+            status_update_rate: If "auto", Queue will send status estimations to all clients whenever a job is finished. Otherwise Queue will send status at regular intervals set by this parameter as the number of seconds.
+            client_position_to_load_data: Once a client's position in Queue is less that this value, the Queue will collect the input data from the client. You may make this smaller if clients can send large volumes of data, such as video, since the queued data is stored in memory.
+            default_enabled: If True, all event listeners will use queueing by default.
+        Example:
+            demo = gr.Interface(gr.Textbox(), gr.Image(), image_generator)
```
Comment (abidlabs) [inline gradio/blocks.py:788]: use string shortcuts in Interface for simplicity

File: gradio/blocks.py [hunk 3]
```diff
@@ -768,28 +769,32 @@ def clear(self):
...
-    def configure_queue(
+    @document()
+    def queue(
         self,
-        live_queue_updates: bool = True,
         concurrency_count: int = 1,
-        data_gathering_start: int = 30,
-        update_intervals: int = 5,
-        duration_history_size: int = 100,
+        status_update_rate: float | str = "auto",
```
Comment (abidlabs) [inline gradio/blocks.py:776]: Better to use `float | Literal["auto"]` for the type hint

File: gradio/event_queue.py [hunk 4]
```diff
@@ -31,31 +31,29 @@ class Queue:
...
-        live_queue_updates: bool,
-        queue_concurrency_count: int,
+        live_updates: bool,
+        concurrency_count: int,
         data_gathering_start: int,
         update_intervals: int,
-        duration_history_size: int,
     ):
         """
         See Blocks.configure_queue() docstring for the explanation of parameters.
```
Comment (abidlabs) [inline gradio/event_queue.py:51]:
```suggestion
        See Blocks.queue() docstring for the explanation of parameters.
```

File: guides/1)getting_started/2)key_features.md [hunk 5]
```diff
@@ -119,22 +119,31 @@ with gr.Interface(css="body {background-color: red}") as demo:
...
+demo = gr.Interface(...).queue()
+demo.launch()
 ```
 
 With `Blocks`:
 ```python
 with gr.Blocks() as demo:
     #...
+demo.queue()
 demo.launch(enable_queue=True)
```
Comment (abidlabs) [inline guides/1)getting_started/2)key_features.md:135]:
```suggestion
demo.launch()
```

**Comment by abidlabs** [comment]
Small nits, but overall LGTM! Very clean

File: gradio/blocks.py (hunk 2 above)
Comment (FarukOzderim) [reply gradio/blocks.py:788]: I think we should not use string shortcuts cuz we removed them from our docs and we just support backwards compatibility.

**Review by FarukOzderim — APPROVED** [review]
Documentation and naming for queue function is miles ahead of mine congrats 🎉
The only point I could make is since we change duration history total and count and calculation why don't we take the average or calculate estimations function index by function index, there was an issue around it and could be great if you want to handle it right now which would make our etas more precise.

**Comment by FarukOzderim** [comment]
By the way if we don't use last x number of durations for calculating ETA the calculation could get ugly when the queue size gets very big because when the queue size is Big there's a lot of communication going on which will slow the processes of events.

File: gradio/blocks.py (hunk 2 above)
Comment (aliabid94) [reply gradio/blocks.py:788]: Agree with @FarukOzderim.

**Comment by aliabid94** [comment]
Let's bring in function specific ETA's later.

**Comment by aliabid94** [comment]
> By the way if we don't use last x number of durations for calculating ETA the calculation could get ugly when the queue size gets very big because when the queue size is Big there's a lot of communication going on which will slow the processes of events.

I don't understand why things would be any different with "last x" calculation?

File: gradio/blocks.py (hunk 3 above)
Comment (aliabid94) [reply gradio/blocks.py:785]: not supported in 3.7

**Pull request merged**